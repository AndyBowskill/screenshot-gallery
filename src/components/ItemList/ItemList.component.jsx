import React from 'react';
import Item from '../Item/Item.component';
import './ItemList.style.css';

const ItemList = (props) => {
  return (
    <div className='itemlist'>
      {props.items.map((item) => (
        <Item key={item.id} item={item} />
      ))}
    </div>
  );
};

export default ItemList;
