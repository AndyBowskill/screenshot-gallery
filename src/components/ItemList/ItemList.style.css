.itemlist {
  width: 85vw;
  margin: 1em auto 1em auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 20px;
}
