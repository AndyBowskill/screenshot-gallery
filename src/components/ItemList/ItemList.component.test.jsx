import React from 'react';
import renderer from 'react-test-renderer';
import ItemList from './ItemList.component';

const items = [
  {
    url: 'www.amazon.co.uk',
    screenshot: '',
  },
  {
    url: 'www.twitter.com',
    screenshot: '',
  },
];

test('ItemList component displays correctly.', () => {
  const component = renderer.create(<ItemList items={items} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
