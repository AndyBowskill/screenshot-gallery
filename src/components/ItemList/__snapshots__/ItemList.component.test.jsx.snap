// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList component displays correctly. 1`] = `
<div
  className="itemlist"
>
  <div
    className="item"
  >
    <img
      alt="www.amazon.co.uk"
      height="169px"
      src=""
      width="300px"
    />
    <p>
      www.amazon.co.uk
    </p>
  </div>
  <div
    className="item"
  >
    <img
      alt="www.twitter.com"
      height="169px"
      src=""
      width="300px"
    />
    <p>
      www.twitter.com
    </p>
  </div>
</div>
`;
